module/zfs/zfs_vnops.c

```diff
@@ -409,7 +409,7 @@ zfs_write(znode_t *zp, uio_t *uio, int ioflag, cred_t *cr)
 		lr = zfs_rangelock_enter(&zp->z_rangelock, woff, n, RL_WRITER);
 	}
 
-	if (zn_rlimit_fsize(zp, uio, uio->uio_td)) {
+	if (zn_rlimit_fsize(zp, GET_UIO_STRUCT(uio), uio_td(uio))) {
```

Comment:
The `zn_rlimit_fsize` wrapper could be updated to take the generic `zfs_uio_t *` here to keep the `GET_UIO_STRUCT` out of the common code.

Reply:
I updated this so we no longer call GET_UIO_STRUCT here.